cns/restserver/restserver.go

```diff
@@ -2055,3 +2122,143 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ncSnapshotIntervalInM
 		}
 	}
 }
+
+// isNCWaitingForUpdate :- Determine whether NC version on NMA matches programmed version
+func (service *HTTPRestService) isNCWaitingForUpdate(ncVersion, ncid string) (waitingForUpdate bool, returnCode int, message string) {
+	getNCVersionURL, ok := ncVersionURLs.Load(ncid)
+	if !ok {
+		returnCode = NotFound
+		message = fmt.Sprintf("[Azure-CNS] Network container %s not found", ncid)
+		return
+	}
+
+	response, err := nmagentclient.GetNetworkContainerVersion(ncid, getNCVersionURL.(string))
+	if err == nil {
+		if response.StatusCode == http.StatusOK {
+			var versionResponse nmagentclient.NetworkContainerResponse
+			rBytes, _ := ioutil.ReadAll(response.Body)
+			json.Unmarshal(rBytes, &versionResponse)
+			if versionResponse.ResponseCode == "200" {
+				programmedVersion, _ := strconv.Atoi(ncVersion)
```

Comment:
programmedVersion means 'programmed vfp' -> this is the one nma has. You meant expectedVersion?

Reply:
programmed = published version

NMA needs to sync with published version